### ISSUE TITLE
[Bug]: Rejected by apple because of ITMS-91064

### ISSUE THREAD

**Issue by idamaeng** [braze-inc/braze-unity-sdk#105]
### Braze Unity SDK Version

6.0.0

### Steps To Reproduce

Send app into apple review process and get rejected

### Expected Behavior

Send app into apple review process and not get rejected

### Actual Incorrect Behavior

ITMS-91064: Invalid tracking information - A PrivacyInfo.xcprivacy file contains invalid tracking information at the following path: “Frameworks/BrazeKit.framework/PrivacyInfo.xcprivacy”. NSPrivacyTracking must be true if NSPrivacyTrackingDomains isn‘t empty. Keys and values in your app’s privacy manifest must be valid. For more details about privacy manifest files, visit: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files.

**Comment by jerielng** [comment]
Hi @idamaeng, could you double-check your SDK version to verify that you are on the `6.0.0` release? This issue should be resolved in that version, as there are no longer any privacy tracking domains in the BrazeKit manifest.

**Comment by idamaeng** [comment]
ok now i remember why i went with version 5.2.1 because when using 6.0.0 during signing in xcode i ran into following issue:

<img width="366" alt="Screenshot 2024-05-28 at 16 07 08" src="https://github.com/braze-inc/braze-unity-sdk/assets/42933713/9e0eafe2-cd2c-4eb2-a526-4655a692f014">

do you have any idea how to solve that?

**Comment by jerielng** [comment]
Our [6.0.0 release](https://github.com/braze-inc/braze-unity-sdk/releases) includes some new configurations around the project's `.entitlements` file to allow for further customizations since some projects may use a custom name that doesn't match the default that our project uses.

These changes should be non-breaking, but to get a better idea of your project, could you share how you are using entitlements in your project? It's possible you may need to specify the path for your existing file (the changelog notes should detail some of that process).

**Comment by jerielng** [comment]
Hey @idamaeng, this issue may take a deeper investigation to figure out the cause, but we've filed an internal ticket to take a closer look at this.

If possible, would you be able to share the project folder structure and the contents of your `.entitlements` files before the update and after the update, so we can get a better idea of the comparison between the two states? You can send those in to our support team at support@braze.com

**Comment by idamaeng** [comment]
Thank you for your feedback. Going to send the folder structure before and after update to the mail you mentioned. Do I need to add further info to that mail so the guys now what the mail is about? 

Further is there any workaround to pass apples review process with 5.2.1 or 6.0.0 its very urgent because we need to release and update to our apps which all utilize the braze sdk.

**Comment by jerielng** [comment]
Hey @idamaeng, we've received your files, thanks! Taking a look, it seems that both of your entitlements files are getting merged into a single one under `Unity-iPhone.entitlements`. You should be able to work around this by specifying the path to this file via the `Entitlements File Path` setting in the Braze Configuration window (documentation [here](https://www.braze.com/docs/developer_guide/platform_integration_guides/unity/sdk_integration/ios/)).

More simply, in your Xcode project, it looks like it's listing the reference to both files as a single name: `<your-app.entitlements Unity-iPhone.entitlements`. Xcode is reading this as a single file with the space included and isn't able to find it because those names are getting concatenated somehow. Because these files are merged with all of the contents, the build setting in Xcode should be updated to simply read `Unity-iPhone.entitlements`.

**Comment by idamaeng** [comment]
I think I found out what the issue is with the entitlement files. We updated our Unity version to the latest 2021 release to utilize Unity's privacy manifest integration. Since this update Apple sign in seems to put its entitlement file into the root of the Xcode project. Because I wasn't aware of this behavior and your changes how entitlements are handled. I got a bit confused. Via code I am now setting the entitlement path to the path where the Apple sign in entitlements file is and now merging works automatically as expected. I can build again. I already sent a new version of our app including Braze SDK version 6.0.0 into Apple review. Fingers crossed that it now won't get rejected. After being successfully reviewed or being rejected because non Braze related issues I am going to close this ticket.

Thanks for your quick replies and help.
